test/clj/rems/application/test_model.clj

```diff
@@ -661,8 +680,12 @@
                                                :application/todo :resubmitted-application
                                                ;; when there was no draft-saved event, the current and
                                                ;; previous submitted answers must be the same
-                                               :rems.application.model/submitted-answers {"41" "foo" "42" "bar" "43" "private answer"}
-                                               :rems.application.model/previous-submitted-answers {"41" "foo" "42" "bar" "43" "private answer"}}))]
+                                               :rems.application.model/submitted-answers [{:form 40 :field "41" :value "foo"}
+                                                                                          {:form 40 :field "42" :value "bar"}
+                                                                                          {:form 40 :field "43" :value "private answer"}]
+                                               :rems.application.model/previous-submitted-answers [{:form 40 :field "41" :value "foo"}
+                                                                                                   {:form 40 :field "42" :value "bar"}
+                                                                                                   {:form 40 :field "43" :value "private answer"}]}))]
           (is (= expected-application (recreate expected-application))))))))
 
 (deftest test-application-view-resources-changed
```

Comment:
I'd like to see a test for `:application/forms` changing when resources are changed

Reply:
Yeah that part is not quite supported yet. Obviously changing ~~resources~~catalogue-items so that there is now extra forms should cause the forms appear and answers not disappear. Maybe in the submit we can clean up the old answers.

Reply:
I think we can just leave the old answers lying around in the model, they won't hurt

Reply:
how about just adding a test that documents the current behaviour for now? as long as the behaviour isn't completely broken

Reply:
This should be handled now  as the feature is implemented and tested.